Makefile

```diff
@@ -16,7 +16,7 @@ GOFLAGS=-ldflags "-X github.com/edgexfoundry/device-sdk-go/v2.Version=$(VERSION)
 GIT_SHA=$(shell git rev-parse HEAD)
 
 tidy:
-	go mod tidy
+	go mod tidy -compat=1.17
```

Comment:
The YAML v2 module is causing the need for this. @cloudxxx8 , I thought we moved to only using YAML v3?
I am ok with this for now and remove option once resolved the dependency.

Reply:
It has nothing to do with YAML v3.  See https://go.dev/ref/mod#graph-pruning.  It has to do with how go generates go.mod and go.sum because by default go 1.17 does dynamic on-the-fly dependency walking but go 1.16 didn't, and go 1.17 modules by default want to be compilable on 1.16 as well.  So 1.17 is a transitional period.  Its complicated.

Reply:
This is the new error we have been getting recently well after changing to Go 1.17
```
        gopkg.in/yaml.v2 loaded from gopkg.in/yaml.v2@v2.2.8,
        but go 1.16 would select v2.3.0
```

Reply:
No CVE's for either version.  yaml.v2 2.3 introduced via go-kit/kit -> etcd/client -> promentheus/client_golang -> prometheus/common